File: spec/challenge.spec.js
'use strict';

const romanNumerals = require('../lib/challenge.js');

describe('Challenge: ', () => {

  describe('arabicToRoman', () => {
    const arabicToRoman = romanNumerals.arabicToRoman;
    console.log(romanNumerals);

    it(`produces simple roman numerals`, () => {
      [ [3,'III'],
        [20,'XX'],
        [100,'C']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(arabicToRoman(arabic)).toEqual(roman);
      });
    });

    it(`produces non-subtractive roman numerals`, () => {
      [ [15, 'XV'],
        [6, 'VI'],
        [123, 'CXXIII'],
        [71, 'LXXI']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(arabicToRoman(arabic)).toEqual(roman);
      });
    });

    it(`produces subtractive roman numerals`, () => {
      [ [4, 'IV'],
        [9, 'IX'],
        [99, 'XCIX'],
        [405, 'CDV']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(arabicToRoman(arabic)).toEqual(roman);
      });
    });
  });

  xdescribe('romanToArabic', () => {
    const romanToArabic = romanNumerals.romanToArabic;

    it(`handles simple roman numerals`, () => {
      [ [3,'III'],
        [20,'XX'],
        [100,'C']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(romanToArabic(roman)).toEqual(arabic);
      });
    });

    it(`handles non-subtractive roman numerals`, () => {
      [ [15, 'XV'],
        [6, 'VI'],
        [123, 'CXXIII'],
        [71, 'LXXI']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(romanToArabic(roman)).toEqual(arabic);
      });
    });

    it(`handles subtractive roman numerals`, () => {
      [ [4, 'IV'],
        [9, 'IX'],
        [99, 'XCIX'],
        [405, 'CDV']].forEach((pair)=>{
        let arabic = pair[0];
        let roman = pair[1];
        expect(romanToArabic(roman)).toEqual(arabic);
      });
    });
  });

});

File: spec/challenge_spec.rb
require_relative '../lib/challenge.rb'

RSpec.describe "Challenge: " do

  describe 'arabic_to_roman' do
    it 'produces simple roman numerals' do
      [ [3,'III'],
        [20,'XX'],
        [100,'C']].each do | arabic, roman |
        expect(arabic_to_roman(arabic)).to eql(roman);
      end
    end

    it 'produces non-subtractive roman numerals' do
      [ [15, 'XV'],
        [6, 'VI'],
        [123, 'CXXIII'],
        [71, 'LXXI']].each do | arabic, roman |
        expect(arabic_to_roman(arabic)).to eql(roman);
      end
    end

    it 'produces subtractive roman numerals' do
      [ [4, 'IV'],
        [9, 'IX'],
        [99, 'XCIX'],
        [405, 'CDV']].each do | arabic, roman |
        expect(arabic_to_roman(arabic)).to eql(roman);
      end
    end
  end

  xdescribe 'roman_to_arabic' do
    it 'converts simple roman numerals into arabic numbers' do
      [ [3,'III'],
        [20,'XX'],
        [100,'C']].each do | arabic, roman |
        expect(roman_to_arabic(roman)).to eql(arabic);
      end
    end

    it 'converts non-subtractive roman numerals into arabic numbers' do
      [ [15, 'XV'],
        [6, 'VI'],
        [123, 'CXXIII'],
        [71, 'LXXI']].each do | arabic, roman |
        expect(roman_to_arabic(roman)).to eql(arabic);
      end
    end

    it 'converts subtractive roman numerals into arabic numbers' do
      [ [4, 'IV'],
        [9, 'IX'],
        [99, 'XCIX'],
        [405, 'CDV']].each do | arabic, roman |
        expect(roman_to_arabic(roman)).to eql(arabic);
      end
    end
  end

end
